pmm/pmm2-server-release.groovy

```diff
@@ -108,43 +111,56 @@ pipeline {
         stage('Set Tags') {
             steps {
                 withCredentials([string(credentialsId: 'GITHUB_API_TOKEN', variable: 'GITHUB_API_TOKEN')]) {
-                    unstash 'copy'
-                    sh """
-                        echo ${GITHUB_API_TOKEN} > GITHUB_API_TOKEN
-                        echo ${VERSION} > VERSION
-                    """
-                    sh '''
-                        export VERSION=$(cat VERSION)
-                        export TOP_VER=$(cat VERSION | cut -d. -f1)
-                        export MID_VER=$(cat VERSION | cut -d. -f2)
-                        export DOCKER_MID="$TOP_VER.$MID_VER"
-                        declare -A repo=(
-                            ["percona-dashboards"]="percona/grafana-dashboards"
-                            ["pmm-server"]="percona/pmm-server"
-                            ["percona-qan-api2"]="percona/qan-api2"
-                            ["pmm-update"]="percona/pmm-update"
-                            ["pmm-managed"]="percona/pmm-managed"
-                        )
-
-                        for package in "${!repo[@]}"; do
-                            SHA=$(
-                                grep "^$package-$VERSION-" copy.list \
-                                    | perl -p -e 's/.*[.]\\d{10}[.]([0-9a-f]{7})[.]el7.*/$1/'
+                    sshagent(['jenkins-github']) {
+                        unstash 'copy'
+                        sh """
+                            echo ${GITHUB_API_TOKEN} > GITHUB_API_TOKEN
+                            echo ${VERSION} > VERSION
+                        """
+                        sh '''
+                            export GPG_AGENT_INFO=${HOME}/.gnupg/S.gpg-agent
+                            export VERSION=$(cat VERSION)
+                            export TOP_VER=$(cat VERSION | cut -d. -f1)
+                            export MID_VER=$(cat VERSION | cut -d. -f2)
+                            export DOCKER_MID="$TOP_VER.$MID_VER"
+                            declare -A repo=(
+                                ["percona-dashboards"]="percona/grafana-dashboards"
+                                ["pmm"]="percona/pmm"
+                                ["pmm-admin"]="percona/pmm-admin"
+                                ["pmm-agent"]="percona/pmm-agent"
+                                ["pmm-server"]="percona/pmm-server"
+                                ["pmm-server-packaging"]="percona/pmm-server-packaging"
+                                ["percona-qan-app"]="percona/qan-app"
+                                ["percona-qan-api2"]="percona/qan-api2"
+                                ["pmm-update"]="percona/pmm-update"
+                                ["pmm-managed"]="percona/pmm-managed"
```

Comment:
Please include QA Repo also https://github.com/percona/pmm-qa

Reply:
This would help us to use tests with version tags to run for a specific version.